Tests/Foundation/Tests/TestAffineTransform.swift

```diff
@@ -7,383 +7,741 @@
-        XCTAssertEqual(Double(transformStruct.tX), Double(0), accuracy: accuracyThreshold)
-        XCTAssertEqual(Double(transformStruct.tY), Double(0), accuracy: accuracyThreshold)
+// MARK: - Construction
+
+extension TestAffineTransform {
+    func testConstruction() {
+        let transform = AffineTransform(
+            m11: 1, m12: 2,
+            m21: 3, m22: 4,
+             tX: 5,  tY: 6
+        )
+        
+        XCTAssertEqual(transform.m11, 1)
+        XCTAssertEqual(transform.m12, 2)
+        XCTAssertEqual(transform.m21, 3)
+        XCTAssertEqual(transform.m22, 4)
+        XCTAssertEqual(transform.tX , 5)
+        XCTAssertEqual(transform.tY , 6)
     }
+}
 
-    func test_IdentityTransformation() {
-        let identityTransform = NSAffineTransform()
+// MARK: - Bridging
 
-        func checkIdentityPointTransformation(_ point: NSPoint) {
-            checkPointTransformation(identityTransform, point: point, expectedPoint: point)
-        }
+extension TestAffineTransform {
+    func testBridging() {
+        let transform = AffineTransform(
+            m11: 1, m12: 2,
+            m21: 3, m22: 4,
+             tX: 5,  tY: 6
+        )
         
-        checkIdentityPointTransformation(NSPoint.zero)
-        checkIdentityPointTransformation(NSPoint(x: 24.5, y: 10.0))
-        checkIdentityPointTransformation(NSPoint(x: -7.5, y: 2.0))
+        let nsTransform = NSAffineTransform(transform: transform)
+        XCTAssertEqual(transform, nsTransform.affineTransform)
+        
+        XCTAssertEqual(nsTransform as AffineTransform, transform)
+    }
+}
 
-        func checkIdentitySizeTransformation(_ size: NSSize) {
-            checkSizeTransformation(identityTransform, size: size, expectedSize: size)
+// MARK: Equality
+
+extension TestAffineTransform {
+    func testEqualityHashing() {
+        let samples = [
+            AffineTransform(m11: 1.5, m12: 2.0, m21: 3.0, m22: 4.0, tX: 5.0, tY: 6.0),
+            AffineTransform(m11: 1.0, m12: 2.5, m21: 3.0, m22: 4.0, tX: 5.0, tY: 6.0),
+            AffineTransform(m11: 1.0, m12: 2.0, m21: 3.5, m22: 4.0, tX: 5.0, tY: 6.0),
+            AffineTransform(m11: 1.0, m12: 2.0, m21: 3.0, m22: 4.5, tX: 5.0, tY: 6.0),
+            AffineTransform(m11: 1.0, m12: 2.0, m21: 3.0, m22: 4.0, tX: 5.5, tY: 6.0),
+            AffineTransform(m11: 1.0, m12: 2.0, m21: 3.0, m22: 4.0, tX: 5.0, tY: 6.5),
+        ].map(NSAffineTransform.init)
+        
+        for (index, sample) in samples.enumerated() {
+            let otherSamples: [NSAffineTransform] = {
+                var samplesCopy = samples
+                samplesCopy.remove(at: index)
+                return samplesCopy
+            }()
+            
+            XCTAssertEqual(sample, sample)
+            XCTAssertEqual(sample.hashValue, sample.hashValue)
+            
+            for otherSample in otherSamples {
+                XCTAssert(sample != otherSample)
+                XCTAssert(sample.hashValue != otherSample.hashValue)
```

Comment:
Might be better to use `XCTAssertNotEqual()` insert to `XCTAssert` as if there is a test failure it will show the incorrect values.